File: components/credentials/src/main/java/org/cloudfoundry/credhub/requests/UsernameValue.java
package org.cloudfoundry.credhub.requests;

import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;

public class UsernameValue {
  private String username;

  public String getUsername() {
    return username;
  }

  public void setUsername(final String username) {
    this.username = username;
  }

  @Override
  public boolean equals(final Object o) {
    if (this == o) {
      return true;
    }

    if (o == null || getClass() != o.getClass()) {
      return false;
    }

    final UsernameValue that = (UsernameValue) o;

    return new EqualsBuilder()
      .append(username, that.username)
      .isEquals();
  }

  @Override
  public int hashCode() {
    return new HashCodeBuilder(17, 37)
      .append(username)
      .toHashCode();
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/requests/UsernameValue.kt
package org.cloudfoundry.credhub.requests

import org.apache.commons.lang3.builder.EqualsBuilder
import org.apache.commons.lang3.builder.HashCodeBuilder

class UsernameValue {
    var username: String? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }

        if (other == null || javaClass != other.javaClass) {
            return false
        }

        val that = other as UsernameValue?

        return EqualsBuilder()
            .append(username, that!!.username)
            .isEquals
    }

    override fun hashCode(): Int =
        HashCodeBuilder(17, 37)
            .append(username)
            .toHashCode()
}
